lib/pending_log_event.ex

```diff
@@ -10,5 +10,42 @@ defmodule LogflareLogger.PendingLoggerEvent do
   def changeset(struct, params) do
     struct
     |> cast(params, [:body, :api_request_started_at])
+    |> update_change(:body, &fix_body/1)
   end
+
+  defp fix_body(change) do
+    change
+    |> Enum.map(fn {k, v} -> {k, check_deep_struct(v)} end)
+    |> Map.new()
+  end
+
+  defp check_deep_struct(v) when is_map(v) do
+    v
+    |> Enum.map(fn {k, v} -> {k, check_deep_struct(v)} end)
+    |> Map.new()
+  end
+
+  defp check_deep_struct(v) when is_list(v) do
+    single_type? =
+      v
+      |> Enum.map(&type/1)
+      |> Enum.uniq()
+      |> then(&(length(&1) == 1))
+
+    case single_type? do
+      true -> v
+      false -> Enum.map(v, &inspect(&1))
```

Comment:
this will stringify the entire list as is and nested maps, and the result would not be json encoded either.
I think doing such large changes to the raw payload isn't such a good idea, and converting the ip/port format into a map as discussed would be a better approach. it would be easy to pattern match on the ipv4 format as well.

Reply:
It's more than the IP problem, it's also the plug name and the full conn map.

I can do a quick pattern match for this specific scenario and handle it since it was the one that triggers it

Reply:
In fact, any kind of Exception that is raised and has a list with different types in the data sent would be broken. 

In this scenario the plugs blows up, uses Logger.error, we capture it and then fail handling it but this could happen with any lib 😅

Reply:
There are other data structures we have to do custom things for e.g. [tuples](https://github.com/Logflare/logflare_logger_backend/blob/master/README.md?plain=1#L66). 

I wouldn't be opposed to stringify-ing all elements in lists with multiple types here. At least it's there then and people could query it reasonably well with regex.

Anecdotally, I've only come across it with this Plug.Conn struct. 

I am wondering if we should run those through Jason though vs inspect.

Reply:
Json encoding it had the added benefit of being compatible with BQ json functions.
We could add some lql operators for json search to ease the pain of regex searching with the character escaping of the json encoded struct.